### DISCUSSION TITLE
Json schema reference to other file

### DISCUSSION THREAD

**Discussion by AdamOndrejkovic** [json-schema-org/community#107] [Q&A] [answered]
I wanted to ask for help regarding a JSON schema that I have in which I am trying to make reference to subschema which is a different file so that the code doesn't get too long. I've looked for some references but none of them worked.
Main:
![main](https://user-images.githubusercontent.com/56413320/140539407-a425173a-a0b2-4b9d-81dd-ff88ea6cd38d.png)
SubMain:
![sub](https://user-images.githubusercontent.com/56413320/140539392-64fb2de6-4587-4a32-996d-d4c95ec6d290.png)

**Comment by karenetheridge** [comment, marked as answer]
What is the question?

**Reply by AdamOndrejkovic** [reply]
The question is how can I reference from the first file mainSchema.json in my array of steps and items refer to submain.json

**Reply by karenetheridge** [reply]
The `$ref` keyword is described here: https://json-schema.org/understanding-json-schema/structuring.html#ref

What you've got so far is fine -- you just need to add an `$id` keyword to the file you are using in `$ref`.

**Reply by AdamOndrejkovic** [reply]
So basically I need to add in my subschema `"$id": "subschema"` and then the ref should refer to that from the main file `"$ref": "subschema.json"` because when I tried it I got an error

**Reply by AdamOndrejkovic** [reply]
This is an error I get __can't resolve reference steps_schema.json from id #_
_

**Reply by karenetheridge** [reply]
"subschema" != "subschema.json".

**Reply by AdamOndrejkovic** [reply]
Even when I change them to be same I get an error

**Reply by karenetheridge** [reply]
What error?

Have you told the evaluator about both files? Evaluators are not required to be able to load files from disk or the network - you may have to introduce the files to the evaluator first before it is able to reference them.

**Reply by AdamOndrejkovic** [reply]
The same error as before

**Reply by AdamOndrejkovic** [reply]
I got it working after using ajv with add schema. Can I add as many schemas as I want and also split them so that the code is kept clean?

**Reply by karenetheridge** [reply]
I don't see why not, unless the implementation imposes an artificial limit.
